app/workers/scheduler/counting_scheduler.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class Scheduler::CountingScheduler
+  include Sidekiq::Worker
+
+  sidekiq_options unique: :until_executed, retry: 0
+
+  def perform
+    recount_tag_scores!
+  end
+
+  private
+
+  def recount_tag_scores!
+    Tag.find_each { |tag| tag.update(score: Redis.current.pfcount(*(0..7).map { |i| "activity:tags:#{tag.id}:#{i.days.ago.beginning_of_day.to_i}:accounts" })) }
```

Comment:
Isn't that going to be a *lot* of queries?
On my small single-user instance alone, there are 37626 known tags.

Reply:
While there will be 37626 fetches from Redis, in the new code an update will only be issued if the numbers differ. If both are 0, such as in the likely majority of cases, the only tags that will be updated will be the ones currently in use.

Reply:
I tried this code on my instance (taruntarun.net).
(The result of Tag.count is 56109.)

I ran CountingScheduler by pressing the “Enqueue now” button on Sidekiq.
The first run took 3 minutes to finish.
The second run was 1 minute.
However, CPU usage increased about 50% during that time.

Reply:
I think 3 minutes of 50% CPU is acceptable? I suspect that it's the Redis pfcount-with-merge operation causing the CPU spike, but I do not have a solution for it.

Reply:
(Personal opinion) I think it is acceptable.
This is less than my server scheduled `pg_dump`'s CPU usage.